datalad_gooey/app.py

```diff
@@ -295,6 +302,39 @@ def _get_info(self):
               'datalad.org</a>.'
         mbox(self.main_window, title, msg)
 
+    def _get_diagnostic_info(self):
+        self.execute_dataladcmd.emit(
+            'wtf',
+            MappingProxyType(dict(
+                result_renderer='disabled',
+                on_failure='ignore',
+                return_type='generator',
+            )),
+            MappingProxyType(dict())
+        )
+        self._cmdexec.results_received.connect(
+            self._app_cmdexec_results_handler)
+
+    @Slot(Interface, list)
+    def _app_cmdexec_results_handler(self, cls, res):
+        for r in res:
+            self._wtf_result_receiver(r)
+
+    def _wtf_result_receiver(self, res):
+        if not res['action'] == 'wtf':
+            return
```

Comment:
This needs another condition. Otherwise any `wtf` call will end up here. Check how `FSBrowser` does this:

It injects a custom result override, that becomes part of any command result

https://github.com/datalad/datalad-gooey/blob/e5c8ba1442e963bd3a47906040fa1e39e47c225e/datalad_gooey/fsbrowser.py#L110-L115

The can then be tested for in the result receiver

https://github.com/datalad/datalad-gooey/blob/e5c8ba1442e963bd3a47906040fa1e39e47c225e/datalad_gooey/fsbrowser.py#L140-L147

any result that doesn't have the secret handshake, can be rejected.

So oit is a two-step rejection:

1. Any non-`wtf` are rejected right away in ` _app_cmdexec_results_handler`
2. Any `wtf` result without the secret stamp are rejected in `_wtf_result_receiver`

Reply:
ahh thats clever, thx!